pkg/cloud/config.go

```diff
@@ -17,6 +17,7 @@ type Config interface {
 type KeyedConfig interface {
 	Config
 	Key() string
+	Provider() string
```

Comment:
Adding this function to the interface allows to removal of the GetProviderFromIntegration helper function which needed to be updated elsewhere